ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -81,6 +89,21 @@ let ntlm_auth uname passwd : (unit, exn) result =
     Ok ()
   with _ -> Error (auth_ex uname)
 
+let get_domain_info_from_db () =
+  (fun __context ->
+    let host = Helpers.get_localhost ~__context in
```

Comment:
Why is this explicitly curried and `__context` not an ordinary parameter?

Reply:
because there are many places the callers do not have a valid `__context`